src/amltk/pipeline/node.py

```diff
@@ -776,3 +865,508 @@ def _fufill_param_requests(
                     continue
 
         return new_config
+
+    @overload
```

Comment:
Mhm, I really do not like overload anymore after seeing this. 
Do you know if Python plans to move on or get some better syntax for this? Or does something better exist that does the same (for mypy)?